test/Microsoft.Sbom.Targets.Tests/GenerateSbomE2ETests.cs

```diff
@@ -0,0 +1,458 @@
+        catch (Exception ex)
+        {
+            Assert.Fail($"Failed to cleanup output directories. {ex}");
+        }
+    }
+
+    private void GetDefaultProperties(Project sampleProject)
+    {
+        expectedPackageName = sampleProject.GetPropertyValue("PackageId");
+        expectedVersion = sampleProject.GetPropertyValue("Version");
+        expectedSupplier = sampleProject.GetPropertyValue("Authors");
+        assemblyName = sampleProject.GetPropertyValue("AssemblyName");
+        configuration = sampleProject.GetPropertyValue("Configuration");
+
+        if (expectedPackageName.IsNullOrEmpty())
+        {
+            expectedPackageName = assemblyName;
+        }
+
+        if (expectedSupplier.IsNullOrEmpty())
+        {
+            expectedSupplier = assemblyName;
+        }
+
+        if (expectedVersion.IsNullOrEmpty())
+        {
+            expectedVersion = "1.0.0";
+        }
+
+        expectedNamespace = $"http://spdx.org/spdxdocs/{expectedPackageName}";
+    }
+
+    private void RestoreBuildPack(Project sampleProject)
+    {
+        var logger = new ConsoleLogger();
+
+        // Restore the project to create project.assets.json file
+        var restore = sampleProject.Build("Restore", new[] { logger });
+        Assert.IsTrue(restore, "Failed to restore the project");
+
+        // Next, build the project
+        var build = sampleProject.Build(logger);
+        Assert.IsTrue(build, "Failed to build the project");
+
+        // Finally, pack the project
+        var pack = sampleProject.Build("Pack", new[] { logger });
+        Assert.IsTrue(pack, "Failed to pack the project");
+    }
+
+    private void ExtractPackage()
+    {
+        // Unzip the contents of the NuGet package
+        var nupkgPath = Path.Combine(projectDirectory, "bin", configuration);
+        var nupkgFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.nupkg");
+        var zipFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.zip");
+        var extractPath = Path.Combine(projectDirectory, "bin", configuration, $"{Guid.NewGuid()}.temp");
+
+        // Rename the .nupkg file to .zip
+        File.Copy(nupkgFile, zipFile, true);
+
+        // Extract the .zip file
+        ZipFile.ExtractToDirectory(zipFile, extractPath);
+
+        manifestPath = Path.Combine(extractPath, "_manifest", sbomSpecificationDirectoryName, "manifest.spdx.json");
+        buildDropPath = extractPath;
+    }
+
+    [TestMethod]
+    public void SbomGenerationSucceedsForDefaultProperties()
+    {
+        //Arrange
+        // Create a Project object for ProjectSample1
+        var sampleProject = (Project)project;
+
+        // Get all the expected default properties
+        GetDefaultProperties(sampleProject);
+
+        // Set the TargetFrameworks property to empty. By default, it sets this property to net6.0 and net8.0, which fails for net8.0 builds.
+        sampleProject.SetProperty("TargetFrameworks", string.Empty);
```

Comment:
If you need to do these steps for all the test cases, try to bundle them into one set-up method, that returns the `sampleProject`, or sets a private property `this.sampleProject` that then you use in the test cases.

Reply:
Updated!